java/j2ee.eclipselink/nbproject/project.xml

```diff
@@ -26,29 +26,126 @@
             <code-name-base>org.netbeans.modules.j2ee.eclipselink</code-name-base>
             <module-dependencies/>
             <public-packages>
-                <subpackages>org.eclipse</subpackages>
+                <subpackages>org.eclipse.persistence</subpackages>
```

Comment:
What is the reason for this change? Are there internal packages, that shall be hidden? If only a single package should be exposed, I would expect `package`, `subpackages` matches all packages where the given name is a prefix.

Reply:
I was under the impression that I should only include `subpackages` that have java classes inside and exclude internal packages like `org.eclipse.persistence.internal`. What would be the proper way of configure `<public-packages>` and `<subpackages>`, include all packages excluding internal packages?

Reply:
If you want to expose only the package and not the package where it is a prefix of, `package` is the right element. Your assessment is right, internal should not be exposed.

The sigtest fails from my interpretation because classes are missing:

https://app.travis-ci.com/github/apache/netbeans/jobs/560459188#L12806-L12808

This smells like these:
https://search.maven.org/artifact/jakarta.resource/jakarta.resource-api/1.7.4/jar
https://search.maven.org/artifact/jakarta.transaction/jakarta.transaction-api/1.3.3/jar

Reply:
I restore the entry `<subpackages>org.eclipse</subpackages>`, deleted the added entries, regenerate sigtest and everything worked fine. I suppose that if another module need a class from this packages it would fail the build process.